stl/inc/xtree

```diff
@@ -178,7 +178,7 @@ public:
     _NODISCARD reference operator*() const noexcept {
 #if _ITERATOR_DEBUG_LEVEL == 2
         const auto _Mycont = static_cast<const _Mytree*>(this->_Getcont());
-        _STL_ASSERT(_Mycont, "cannot dereference value-initialized map/set iterator");
+        _STL_VERIFY(_Mycont, "cannot dereference value-initialized map/set iterator");

```

Comment:
The other adjusted error messages make me wonder if this should be something like `operator*() called on a value-initialized iterator`.

Reply:
Yeah, that would be consistent.

Unlike the continue-on-error audit, where I went through every single `_STL_VERIFY` / `_STL_ASSERT` / etc. in the STL, for the debug messages I audited only those that are shared by checks that will be part of STL Hardening. (I realize that I didn't mention the scope of this cleanup in the PR description.)

Since I'll be adjusting other messages, I can do a quick review and clean some of these up. There are ~700 such checks in the STL so I don't want to promise a complete cleanup here. :joy_cat:

Reply:
Yeah that seems like a lot of checks.  I commented because this line was modified so it seemed like fair game to ask about.

Reply:
After searching for `_STL_VERIFY|_STL_ASSERT`, there are a ton of messages that could be rephrased. I'll record a personal todo but I'm going to rule that this is out of scope for this PR. I'll adjust the PR description to note the scope here.